cli/internal/core/engine.go

```diff
@@ -385,11 +410,76 @@ func (e *Engine) ValidatePersistentDependencies(graph *graph.CompleteGraph) erro
 	return validationError
 }
 
-// GetResolvedTaskDefinition returns a "resolved" TaskDefinition.
-// Today, it just looks up the task from the root Pipeline, but in the future
-// we will compose the TaskDefinition from workspaces using the `extends` key.
-func (e *Engine) GetResolvedTaskDefinition(rootPipeline *fs.Pipeline, taskName string, taskID string) (*fs.TaskDefinition, error) {
-	return rootPipeline.GetTask(taskID, taskName)
+// getTaskDefinitionChain gets a set of TaskDefinitions that apply to the taskID.
+// These definitions should be merged by the consumer.
+// TODO(mehulkar): Can we use getPipelineFromWorkspace() here instead of reading configs again and passing around rootPipeline as an arg?
+func (e *Engine) getTaskDefinitionChain(rootPipeline *fs.Pipeline, pkg *fs.PackageJSON, taskID string, taskName string) ([]fs.BookkeepingTaskDefinition, error) {
+	// Start a list of TaskDefinitions we've found for this TaskID
+	taskDefinitions := []fs.BookkeepingTaskDefinition{}
+
+	// Look for the taskDefinition in the root pipeline.
+	// We'll wait to throw errors until the end
```

Comment:
I think this comment is either wrong or maybe just in the wrong place. We aren't saving the error from the next line, we're immediately discarding it if it's non-nil.